testrunner-running/src/main/scala/edu/illinois/cs/testrunner/data/framework/Utils.java

```diff
@@ -0,0 +1,25 @@
+package edu.illinois.cs.testrunner.data.framework;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+class Utils {
```

Comment:
I think it's not very suitable to put this method in `TestFramework.scala` file. Plus this method is like a helper method. So I created this class. Do you have a better idea?

Reply:
It's fine if you want to leave `getAllMethods` here. 

Would getDeclaredMethods also work for JUnit 4? If so, then it would be good to make the two consistent by default and perhaps just add a configuration option for JUnit 4 to get tests the current way in case we need to reproduce something later.

Reply:
That should also work for J4. But I don't think that's necessary. JUnit 4 tests must be public. And the `getMethods()` (used for J4 now) will return all public methods including the inherited ones.

So the current way for J4 works expectedly, I don't think we should change it if it's not a bug. But anyway, this is just my thought. We can still change it as you requested if you think it's necessary.

Reply:
If the behavior (return all public methods including the inherited ones) for getting the list of test methods between J4 and J5 (even though the implementation may be different) is the same, then you need not make any changes to the way J4 gets the test methods.

@august782 can you look over the changes as well? I am fine with the changes as is.

Reply:
By the way, is it possible to give me partial permission which allows me to modify the metadata of PR and Issue so that I can link the PR to a certain issue?